packages/react-reconciler/src/ReactFiberViewTransitionComponent.js

```diff
@@ -39,26 +40,61 @@ export function assignViewTransitionAutoName(
 
   const root = ((getWorkInProgressRoot(): any): FiberRoot);
   const identifierPrefix = root.identifierPrefix;
+  let owner = '';
+  if (__DEV__) {
+    if (fiber._debugOwner) {
+      const componentName = getComponentNameFromOwner(fiber._debugOwner);
+      if (componentName) {
+        owner = formatGeneratedName(componentName);
+      } else {
+        owner = 'UnnamedOwner';
+      }
+    } else {
+      owner = 'UnknownOwner';
+    }
+    owner = '_' + owner + '_';
+  }
 
   let name;
   if (getIsHydrating()) {
     const treeId = getTreeId();
-    // Use a captial R prefix for server-generated ids.
-    name = '\u00AB' + identifierPrefix + 'T' + treeId + '\u00BB';
+    // Use a captial T prefix for server-generated ids.
+    name = '\u00AB' + identifierPrefix + 'T' + treeId + owner + '\u00BB';
   } else {
-    // Use a lowercase r prefix for client-generated ids.
+    // Use a lowercase t prefix for client-generated ids.
     const globalClientId = globalClientIdCounter++;
     name =
       '\u00AB' +
       identifierPrefix +
       't' +
       globalClientId.toString(32) +
+      owner +
       '\u00BB';
   }
   instance.autoName = name;
   return name;
 }
 
+// Strip to 16 characters and replace invalid characters.
+// We don't need to replace invalid first characters or forbidden values
+// because result will be prefixed.
+function formatGeneratedName(str: string) {
+  let result = '';
+  // Allow a max of 16 characters
+  for (let i = 0; i < Math.min(16, str.length); i++) {
+    const char = str[i];
+    // Check if the character is a valid CSS ident character
+    if (/[a-zA-Z0-9_-]/.test(char)) {
+      result += char;
+    } else {
+      // Replace invalid characters
+      result += '\u2733';
+    }
+  }
+
+  return result.trim();
```

Comment:
I think this could be just a single regex replace, avoiding a bunch of string concatenations:

```suggestion
  return str.slice(0, 16).replace(/[^a-zA-Z0-9_-]/g, '\u2733');
```

Reply:
Updated